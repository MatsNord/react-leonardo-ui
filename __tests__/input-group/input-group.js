import React from 'react';
import renderer from 'react-test-renderer';

import InputGroup from '../../src/input-group';

test('Empty InputGroup renders correctly', () => {
  const tree = renderer.create(
    <InputGroup />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('InputGroup with items renders correctly', () => {
  const tree = renderer.create(
    <InputGroup>
      <InputGroup.Input value="Value" />
      <InputGroup.Button>Text</InputGroup.Button>
    </InputGroup>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('InputGroup passes props correctly', () => {
  const onClick = jest.fn(() => {});

  const tree = renderer.create(
    <InputGroup className="my-class" title="title" data-x="x" onClick={onClick} />
  ).toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick();
  expect(onClick.mock.calls.length).toBe(1);
});
