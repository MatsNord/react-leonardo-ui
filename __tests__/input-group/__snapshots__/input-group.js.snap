exports[`test Empty InputGroup renders correctly 1`] = `
<div
  className="lui-input-group" />
`;

exports[`test InputGroup passes props correctly 1`] = `
<div
  className="lui-input-group my-class"
  data-x="x"
  onClick={[Function]}
  title="title" />
`;

exports[`test InputGroup with items renders correctly 1`] = `
<div
  className="lui-input-group">
  <input
    className="lui-input lui-input-group__item  lui-input-group__input  "
    value="Value" />
  <button
    className="lui-button lui-input-group__item  lui-input-group__button  ">
    Text
  </button>
</div>
`;
